src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Receivers.kt

```diff
@@ -141,21 +140,26 @@ open class TaskReceiver(tasksDirectory: String,
 }
 
 class MarkdownTaskReceiver(
-        texbeAddress: String,
+        private val texbeCompilerStub: TexbeGrpc.TexbeBlockingStub?,
         tasksDirectory: String,
-        resultPublisher: Publisher) : TaskReceiver(tasksDirectory, resultPublisher) {
+        resultPublisher: PublisherApi,
+        private val config: Config = DEFAULT_CONFIG) : TaskReceiver(tasksDirectory, resultPublisher) {
 
-    private val texbeCompilerStub: TexbeGrpc.TexbeBlockingStub
-
-    init {
-        val channel = ManagedChannelBuilder.forTarget(texbeAddress).usePlaintext(true).build()
-        texbeCompilerStub = TexbeGrpc.newBlockingStub(channel)
-    }
+    private val arguments = listOf("projectRootAbsPath", "workingDirRelPath",
+            "inputFileName", "outputFileName", "mainFont")
+    private val COMPILE_COMMAND_KEY = "pandoc.compile.command"
 
     override fun processMessage(message: PubsubMessage): Boolean {
         val request = CompileRequest.parseFrom(message.data)
+        LOGGER.debug("Converting Markdown to tex: {}", request.mainFileName)
+
         fetchProjectFiles(request)
-        convertMarkdown(request)
+        val commandArguments = getCmdLineArguments(request)
+        val exitCode = convertMarkdown(commandArguments)
+        if (exitCode != 0) {
+            LOGGER.error("Failed to convert Markdown to tex with exitcode {}", exitCode)
+            return false
+        }
 
         val taskId = request.id
         var isPublished = true
```

Comment:
I guess this will never be false in reality because real publishing is asynchronous (that's why callback!), so checking this boolean value in tests doesn't make too much sense.

Also, at line 172 you create mock result bytes. Any sense to actually read the produced output file and send the real output data (and test it of course)?

Reply:
I thought, that compiler-fleet at the moment converts .md to .tex and then sends texbe that converted file, texbe compiles it to .pdf and the final result is publishing by our Publisher. So this mock bytes is result of texbe response. 
Do I need to publish tex result by publisher as well?

Reply:
> compiler-fleet at the moment converts .md to .tex 

I see the point, but since it is not yet useful for production purposes anyway, returning the results of converting will make it more useful in tests, because we'll be able to test more.